### DISCUSSION TITLE
Mixing raw aggregation with eloquent query

### DISCUSSION THREAD

**Discussion by TranLong183** [jenssegers/laravel-mongodb#3200] [Q&A] [answered]
I want to using the raw aggregation with the eloquent supporting method like where(). In my query it perform multiple filter query logic the simple logic I handle with where() but I ran to problem need to use aggregate to implement logic like $findAllRegex, $replaceAll, ... But I have so many where() clause so I dont want to refactor all of this to aggregate then I want to mix it together. Any solution for this

**Comment by alcaeus** [comment, marked as answer]
Raw aggregation and the new aggregation builder are not compatible with `where()`. As aggregation logic is purely specific to MongoDB and not supported by other databases supported by Eloquent, there is no reason for us to support `where()`. It could come in the future, but I can't guarantee it.

Also, if you're using version 4.3.0 or newer, you'll want to check out the [Aggregation Builder](https://www.mongodb.com/docs/drivers/php/laravel-mongodb/current/fundamentals/aggregation-builder/), as it's a lot nicer to use than raw aggregations.

**Reply by TranLong183** [reply]
Thank for your reply, I will check out the Aggregation builder. To be correct it still need to refactor to archive this right.

**Reply by alcaeus** [reply]
Yes, you won't be able to use `where()` in the aggregation builder, but it will be easier for us to support passing a query builder instance to `Stage::match()` in the future than it would be to handle it in `raw()`.

**Reply by TranLong183** [reply]
ok, thank a lot for your reply

**Reply by alcaeus** [reply]
You're too quick for me to edit my own reply. I created PHPORM-262 to track this improvement. Please follow that ticket for updates on this topic.
